os/drivers/lwnl/lwnl.c

```diff
@@ -252,6 +239,17 @@ static int lwnl_poll(FAR struct file *filep, FAR struct pollfd *fds, bool setup)
 
 	/* Are we setting up the poll? Or tearing it down? */
 	if (setup) {
+		/*  Check if any requested events are already in effect */
+		if (fds->events & POLLIN) {
+			int nready = lwnl_check_queue(filep);
+			if (nready > 0) {
+				fds->revents |= (fds->events & POLLIN);
+				sem_post(fds->sem);
+				LWNLDEV_UNLOCK(upper);
```

Comment:
Is it okay to unlock after `sem_post`? Because, sometimes `sem_post` makes context switching directly before calling unlock.